packages/@react-aria/selection/src/useSelectableCollection.ts

```diff
@@ -168,23 +179,33 @@ export function useSelectableCollection(options: SelectableCollectionOptions): S
         break;
       }
       case 'ArrowLeft': {
-        if (delegate.getKeyLeftOf) {
+        if (editModeEnabled) {
+          // Prevent scrolling from happening if in edit mode
+          e.preventDefault();
+        } else if (delegate.getKeyLeftOf) {
           e.preventDefault();
           let nextKey = delegate.getKeyLeftOf(manager.focusedKey);
           navigateToKey(nextKey, direction === 'rtl' ? 'first' : 'last');
         }
         break;
       }
       case 'ArrowRight': {
-        if (delegate.getKeyRightOf) {
+        if (editModeEnabled) {
+          // Prevent scrolling from happening if in edit mode
+          e.preventDefault();
+        } else if (delegate.getKeyRightOf) {
           e.preventDefault();
           let nextKey = delegate.getKeyRightOf(manager.focusedKey);
           navigateToKey(nextKey, direction === 'rtl' ? 'last' : 'first');
         }
         break;
       }
       case 'Home':
-        if (delegate.getFirstKey) {
+        if (editModeEnabled) {
+          // Prevent scrolling from happening if in edit mode
+          // TODO: not sure why this doesn't stop table scrolling when in a searchfield in editmode...
+          e.preventDefault();
+        } else if (delegate.getFirstKey) {
```

Comment:
I believe this can be fixed with an approach like https://github.com/adobe/react-spectrum/pull/1519, but will need to discuss if we want to prevent that behavior on all components that use useTextFields

Reply:
Additionally, scrolling will still happen if the user hits home/end when the cursor is at the start/end of the input text, I figure we will want to suppress that. Open to opinions